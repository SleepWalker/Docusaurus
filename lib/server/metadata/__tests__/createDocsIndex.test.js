/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
const createDocsIndex = require('../createDocsIndex');

const {resolveSidebarItem} = createDocsIndex;

it('should populate docs index from multiple sidebars', () => {
  const result = createDocsIndex({
    docs: {
      Category1: ['doc1', 'doc2'],
      Category2: ['doc3', 'doc4'],
    },
    otherDocs: {
      Category1: ['doc5'],
    },
  });

  expect(result).toMatchSnapshot();
});

it('should resolve docs from older versions', () => {
  const result = createDocsIndex({
    docs: {
      Category1: ['doc1'],
    },
    'version-1.2.3-docs': {
      Category1: ['version-1.2.3-doc2'],
      Category2: ['version-1.2.3-doc1'],
    },
  });

  expect(result).toMatchSnapshot();
});

it('should resolve links to other sidebar docs', () => {
  const result = createDocsIndex({
    docs: {
      Category1: ['doc1'],
    },
    docs2: {
      Category2: ['docs.doc1', 'doc2'],
    },
  });

  expect(result).toMatchSnapshot();
});

describe('resolveSidebarItem()', () => {
  const sidebars = {
    docs: {
      foo: ['installation', 'version-1.3.3-installation'],
    },
  };

  test('simple id', () => {
    const result = resolveSidebarItem('installation', sidebars);

    expect(result).toEqual({
      id: 'installation',
      initialId: 'installation',
      prefix: '',
    });
  });

  test('versioned id', () => {
    const result = resolveSidebarItem('version-1.3.3-installation', sidebars);

    expect(result).toEqual({
      id: 'version-1.3.3-installation',
      initialId: 'version-1.3.3-installation',
      prefix: '',
    });
  });

  test('id from different sidebar', () => {
    const result = resolveSidebarItem('docs.installation', sidebars);

    expect(result).toEqual({
      id: 'installation',
      initialId: 'docs.installation',
      prefix: 'docs',
    });
  });

  test('versioned id from different sidebar', () => {
    const result = resolveSidebarItem(
      'docs.version-1.3.3-installation',
      sidebars
    );

    expect(result).toEqual({
      id: 'version-1.3.3-installation',
      initialId: 'docs.version-1.3.3-installation',
      prefix: 'docs',
    });
  });
});
