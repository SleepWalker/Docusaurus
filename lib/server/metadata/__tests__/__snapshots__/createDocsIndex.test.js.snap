// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should populate docs index from multiple sidebars 1`] = `
Object {
  "doc1": Object {
    "category": "Category1",
    "id": "doc1",
    "next": "doc2",
    "prefix": "",
    "previous": undefined,
    "sidebar": "docs",
  },
  "doc2": Object {
    "category": "Category1",
    "id": "doc2",
    "next": "doc3",
    "prefix": "",
    "previous": "doc1",
    "sidebar": "docs",
  },
  "doc3": Object {
    "category": "Category2",
    "id": "doc3",
    "next": "doc4",
    "prefix": "",
    "previous": "doc2",
    "sidebar": "docs",
  },
  "doc4": Object {
    "category": "Category2",
    "id": "doc4",
    "next": undefined,
    "prefix": "",
    "previous": "doc3",
    "sidebar": "docs",
  },
  "doc5": Object {
    "category": "Category1",
    "id": "doc5",
    "next": undefined,
    "prefix": "",
    "previous": undefined,
    "sidebar": "otherDocs",
  },
}
`;

exports[`should resolve docs from older versions 1`] = `
Object {
  "doc1": Object {
    "category": "Category1",
    "id": "doc1",
    "next": undefined,
    "prefix": "",
    "previous": undefined,
    "sidebar": "docs",
  },
  "version-1.2.3-doc1": Object {
    "category": "Category2",
    "id": "version-1.2.3-doc1",
    "next": undefined,
    "prefix": "",
    "previous": "version-1.2.3-doc2",
    "sidebar": "version-1.2.3-docs",
  },
  "version-1.2.3-doc2": Object {
    "category": "Category1",
    "id": "version-1.2.3-doc2",
    "next": "version-1.2.3-doc1",
    "prefix": "",
    "previous": undefined,
    "sidebar": "version-1.2.3-docs",
  },
}
`;

exports[`should resolve links to other sidebar docs 1`] = `
Object {
  "doc1": Object {
    "category": "Category1",
    "id": "doc1",
    "next": undefined,
    "prefix": "",
    "previous": undefined,
    "sidebar": "docs",
  },
  "doc2": Object {
    "category": "Category2",
    "id": "doc2",
    "next": undefined,
    "prefix": "",
    "previous": "doc1",
    "sidebar": "docs2",
  },
  "docs.doc1": Object {
    "category": "Category2",
    "id": "doc1",
    "next": "doc2",
    "prefix": "docs",
    "previous": undefined,
    "sidebar": "docs2",
  },
}
`;
